src/file.cpp

```diff
@@ -25,16 +25,27 @@
 // Default namespace for Rive Cpp code
 using namespace rive;
 
+#if !defined(RIVE_FMT_U64)
+    #if defined(__ANDROID__)
+        #define RIVE_FMT_U64 "%llu"
+        #define RIVE_FMT_I64 "%lld"
+    #else
+        #include <inttypes.h>
+        #define RIVE_FMT_U64 "%" PRIu64
+        #define RIVE_FMT_I64 "%" PRId64
+    #endif
+#endif
+
 // Import a single Rive runtime object.
 // Used by the file importer.
 static Core* readRuntimeObject(BinaryReader& reader,
                                const RuntimeHeader& header)
 {
-	auto coreObjectKey = reader.readVarUint();
+	auto coreObjectKey = reader.readVarUint64();
```

Comment:
A case where I as a developer couldn't deduce the type by reading the code.

Reply:
I'm always torn about using auto. I got accustomed to it with Dart's var type in combination with their excellent language server integration with VSCODE... the discoverable type info on mouse hover is really nice but you're right that this is not pretty when reading pure text.

Reply:
I think that'd be fine if everyone is using the same IDE, but that will never happen with C/C++.
Also, to hover (for at least 2-300 ms) takes away that amount of time for you, when all you want to do is read the code.

Personally, I don't see a benefit at all. (I know it's a bit anti-modern-c++)
It feels mostly like a way to make it easier to refactor later on. But usually a search-and-replace helps with that.
And, you don't really save time typing. I guess someone might argue about typos, e.g. putting the wrong (legal) type there, but I think that's something that an IDE should help with (and there are also warnings from the compiler).

Sorry for the continued discussion :)

Reply:
No, I think it's great! Happy to change these, there are only a few in the code-base I think...